### PR TITLE
Create option to hardcode metadata in experiment config

### PR DESCRIPTION
**Pull request by zakwatts**
Create option to pass in a hardcode_tilt_orientation true/false statement into the NetcdfPvDataSource class. Which is accessed later on to decide if to use hardcoded tilt and orientation or to use the metadata from the pv file.

### CODE REVIEW

File: psp/exp_configs/uk_pv.py [hunk 1]
```diff
@@ -246,6 +265,12 @@ def get_model(self, *, random_state: np.random.RandomState | None = None) -> PvS
             random_state=random_state,
             normalize_features=True,
             capacity_getter=_get_capacity,
+            tilt_getter=_default_get_tilt
```
Comment (simlmx) [inline psp/exp_configs/uk_pv.py:268]: Why not pass a different `tilt_getter` instead? Why add an option to the data source itself?

File: psp/exp_configs/uk_pv.py [hunk 2]
```diff
@@ -173,6 +173,24 @@ def _get_capacity(d):
...
+    return tilt_values
+
+
+def _get_orientation(d):
+    orientation_values = d["orientation"].values
+    return orientation_values
+
+
+def _default_get_tilt(_):
+    return 35.0
```
Comment (peterdudfield) [inline psp/exp_configs/uk_pv.py:187]: remove

File: psp/data_sources/pv.py [hunk 3]
```diff
@@ -126,6 +128,7 @@ def __init__(
         self._rename = rename
         self._ignore_pv_ids = ignore_pv_ids
         self._lag_minutes = lag_minutes
+        self._hardcode_tilt_orientation = hardcode_tilt_orientation
```
Comment (peterdudfield) [inline psp/data_sources/pv.py:131]: remove

File: psp/exp_configs/uk_pv.py (hunk 1 above)
Comment (peterdudfield) [reply psp/exp_configs/uk_pv.py:268]: yea keep it simple to just `_get_tilt`

File: psp/data_sources/pv.py [hunk 4]
```diff
@@ -110,6 +110,7 @@ def __init__(
                 in practice. Concretely, this means that when we call `as_available_at`,
                 `lag_minutes` minutes will subtracted from the passed timestamp. When training, this
                 should be set to the expected delay before the PV data is available, in production.
+            hardcode_tilt_orientation: If True, we hardcode the tilt and orientation of the PVs.
```
Comment (peterdudfield) [inline psp/data_sources/pv.py:113]: rm

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**